### DISCUSSION TITLE
Cosimulation question of lab1 and lab2

### DISCUSSION THREAD

**Discussion by yining213** [bol-edu/HLS-SOC-Discussions#14] [Q&A]
When I cosimulate projects 'hls_Multiplication' and 'hls_FIRN11MAXI', I got the same error message [COSIM 212-4] and similar INFOs down there. The different is that 'hls_FIRN11MAXI' has  an extra one, which is INFO [COSIM 212-211], but I don't understand the meaning is. How can I solve the problem? 
Thx!

error message in 'hls_Multiplication'
![image](https://user-images.githubusercontent.com/113331483/190868478-9ed46632-d808-4435-b678-d3be563c3659.png)

error message in 'hls_FIRN11MAXI'
![image](https://user-images.githubusercontent.com/113331483/190868494-dc09ea9d-8914-4766-b3e2-c23679b6d8ab.png)

**Comment by tofumanjeff** [comment]
Please provide more detailed information, including your pragmas/directives, synthesis results, and co-sim log files.

**Reply by yining213** [reply]
### **hls_Multiplication:**

- directives: 

![image](https://user-images.githubusercontent.com/113331483/190909734-c6c21fa4-2d08-44e4-9d78-2b8fd9deb62b.png)

- synthesis results:

![圖片3](https://user-images.githubusercontent.com/113331483/190911266-d1d9cd24-9434-4b26-a474-aef1591929de.png)

- co-sim log file:

![image](https://user-images.githubusercontent.com/113331483/190911383-b9fa4a0f-da18-457a-b9bb-58924f10002c.png)

### **hls_FIRN11MAXI:**

- directives: 

![image](https://user-images.githubusercontent.com/113331483/190911921-6f82793b-ab16-4ec9-85ab-b5442963f8ac.png)

- synthesis results:

![圖片4](https://user-images.githubusercontent.com/113331483/190914297-17d4648f-d929-4cd2-9970-b8062302de3e.png)

- co-sim log file:

![image](https://user-images.githubusercontent.com/113331483/190914502-fe99ea12-0535-4779-9046-d9e9d596f10f.png)

**Reply by tofumanjeff** [reply]
I don't see any errors in directives. Did you modify the original code?
Besides, you may trace your console and see if there is any other error message.
There should be no error if you strictly follow the steps in the workbook.

**Reply by yining213** [reply]
I didn't modify any of the code except for the line "#pragma HLS INTERFACE ap_ctrl_none port=return" in directive. I'm sure that I have strictly followed the instruction and there is exactly one error when I cosimulate it. Since the error message tells that 'ap_source' returned error code, is there any other condition that may cause it?

- error in hls_Multiplication:

![image](https://user-images.githubusercontent.com/113331483/191944951-231e54c7-eb04-4957-aadb-920f37bb4662.png)
![image](https://user-images.githubusercontent.com/113331483/191946848-42c0b840-9211-4abc-bd20-7587ca55329e.png)

**Comment by jiinlai** [comment]
Refer to: https://www.boledu.org/textbooks/hls-textbook/application-acceleration-development-flow/crtl-co-simulation#crtl-co-simlation.
and https://github.com/bol-edu/HLS-Discussions/discussions/7
Please check if it helps.

**Reply by yining213** [reply]
Thanks for the suggestions!
